packages/@rescript/runtime/Stdlib_DataView.resi

```diff
@@ -0,0 +1,422 @@
+/* Copyright (C) 2015-2016 Bloomberg Finance L.P.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * In addition to the permissions granted to you by the LGPL, you may combine
+ * or link a "work that uses the Library" with a publicly distributed version
+ * of this file to produce a combined library or application, then distribute
+ * that combined work under the terms of your choosing, with no requirement
+ * to comply with the obligations normally placed on you by section 4 of the
+ * LGPL version 3 (or the corresponding section of a later version of the LGPL
+ * should you choose to use a later version).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. */
+
+/***
+Functions for interacting with DataView.
+
+See [`DataView`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) on MDN.
+*/
+
+/**
+Type representing a DataView.
+*/
+@notUndefined
+type t
+
+/**
+`fromBuffer(buffer)` creates a DataView for the entire ArrayBuffer.
+
+See [`DataView`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) on MDN.
+
+## Examples
+
+```rescript
+let dv = DataView.fromBuffer(buffer)
+```
+*/
+@new external fromBuffer: Stdlib_ArrayBuffer.t => t = "DataView"
+
+/**
+`fromBufferToEnd(buffer, ~byteOffset)` creates a DataView starting at a given byte offset.
+
+See [`DataView`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) on MDN.
+
+## Examples
+
+```rescript
+let dv = DataView.fromBufferToEnd(buffer, ~byteOffset=4)
+```
+*/
+@new external fromBufferToEnd: (Stdlib_ArrayBuffer.t, ~byteOffset: int) => t = "DataView"
```

Comment:
We should consolidate `fromBuffer`, `fromBufferToEnd`, `fromBufferWithRange` into a single `fromBuffer` binding with optional args, deprecating the other two. It seems I forgot this when I previously did the same for `Int8Array`, `Int16Array`, etc.